moped-etl/prefect/tasks/api.py

```diff
@@ -0,0 +1,194 @@
+            "project_name": "atd-moped-test-prefect",
+            "runtime": "python3.8",
+            "s3_bucket": "atd-apigateway",
+            "cors": True,
+            "aws_environment_variables": {
+                "MOPED_API_CURRENT_ENVIRONMENT": "TEST",
+                "MOPED_API_CONFIGURATION_SETTINGS": create_secret_name(basename),
+                "AWS_COGNITO_DYNAMO_TABLE_NAME": AWS_STAGING_DYNAMO_DB_TABLE_NAME,
+                "AWS_COGNITO_DYNAMO_SECRET_NAME": AWS_STAGING_DYNAMO_DB_ENCRYPT_KEY_SECRET_NAME,
+                # Look at Moped API events.py to see how this key is used
+                "MOPED_API_HASURA_APIKEY": MOPED_API_HASURA_APIKEY,
+                "MOPED_API_HASURA_SQS_URL": MOPED_API_HASURA_SQS_URL,
+                "MOPED_API_UPLOADS_S3_BUCKET": MOPED_API_UPLOADS_S3_BUCKET,
+            },
+            "extra_permissions": [
+                {
+                    "Effect": "Allow",
+                    "Action": "secretsmanager:GetSecretValue",
+                    "Resource": [
+                        f"{AWS_STAGING_DYNAMO_DB_ARN}",
+                        config_secret_arn,
+                    ],
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": "cognito-idp:*",
+                    "Resource": f"{AWS_STAGING_COGNITO_USER_POOL_ARN}",
+                },
+            ],
+        }
+    }
+    return zappa_config
+
+
+create_api_task = ShellTask(
+    name="Run API Zappa deploy bash command", stream_output=True, return_all=True
+)
+
+
+@task(name="Create API Zappa deploy bash command")
+def create_moped_api_deploy_command(basename, config_secret_arn):
+    logger.info("Creating API Zappa deploy command")
+
+    zappa_config = create_zappa_config(basename, config_secret_arn)
+    api_project_path = "../../moped-api/"
+
+    # Write Zappa config to moped-api project folder
+    with open(f"{api_project_path}zappa_settings.json", "w") as f:
+        json.dump(zappa_config, f)
+
+    # zappa deploy requires an active virtual environment
+    # then use a subshell to zappa deploy in moped-api project folder
+    command = f"""python3 -m venv venv;
+    . venv/bin/activate;
+    (cd {api_project_path} &&
+    pip install wheel &&
+    pip install -r ./requirements/moped_test.txt &&
+    zappa deploy {basename})
+    deactivate;
+    """
+
+    return command
+
+
+# The output list comes from create_api_task's return_all=True
+# See https://docs.prefect.io/api/latest/tasks/shell.html#shelltask
+@task(name="Get endpoint from Zappa deploy shell task output")
+def get_endpoint_from_deploy_output(output_list):
+    api_endpoint_item = ""
+
+    for item in output_list:
+        if "https://" in item and "amazonaws" in item:
+            api_endpoint_item = item
+
+    match = re.search(r"(https://.*)", api_endpoint_item)
+
+    if match == None:
+        return None
+    else:
+        endpoint = match.groups()[0]
+        return endpoint
```

Comment:
This feels a little brittle since it comes from the command line output of zappa, but I think that the endpoint being in the output is well documented and reliable. The version of zappa is also pinned in the dockerized agent.